template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/efm/EfmRoleConfigProvider.java

```diff
@@ -0,0 +1,56 @@
+package com.sequenceiq.cloudbreak.cmtemplate.configproviders.efm;
+
+import static com.sequenceiq.cloudbreak.cmtemplate.configproviders.ConfigUtils.config;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.stereotype.Component;
+
+import com.cloudera.api.swagger.model.ApiClusterTemplateConfig;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.database.base.DatabaseType;
+import com.sequenceiq.cloudbreak.cmtemplate.configproviders.AbstractRdsRoleConfigProvider;
+import com.sequenceiq.cloudbreak.template.TemplatePreparationObject;
+import com.sequenceiq.cloudbreak.template.views.RdsView;
+
+@Component
+public class EfmRoleConfigProvider extends AbstractRdsRoleConfigProvider {
+
+    static final String DATABASE_USER = "efm.db.username";
+
+    static final String DATABASE_PASSWORD = "efm.db.password";
+
+    static final String DATABASE_URL = "efm.db.url";
+
+    static final String DRIVER_CLASS = "efm.db.driverClass";
+
+    @Override
+    protected List<ApiClusterTemplateConfig> getRoleConfigs(String roleType, TemplatePreparationObject source) {
+        RdsView efmRdsView = getRdsView(source);
+        List<ApiClusterTemplateConfig> configs = new ArrayList<>();
+
+        if (EfmRoles.EFM_SERVER.equals(roleType)) {
```

Comment:
Does not hurt, but also superfluous. `getRoleTypes()` returns `List.of(EfmRoles.EFM_SERVER)`, which means this config provider will only be invoked & evaluated for the appropriate roles. Would only make a difference when multiple roles were supported. Anyway, can be left as is for now.